packages/headless/src/features/insight-user-actions/insight-user-actions-preprocessing.test.ts

```diff
@@ -0,0 +1,745 @@
+import {
+  sortActions,
+  filterActions,
+  mapUserActions,
+  isPartOfTheSameSession,
+  splitActionsIntoTimelineSessions,
+  insertTicketCreationActionInSession,
+  filterTimelineActions,
+  preprocessActionsData,
+} from './insight-user-actions-preprocessing';
+import {UserActionType} from './insight-user-actions-state';
+
+const createRelativeDate = (date: Date, minutes: number, seconds: number) => {
+  const totalSeconds = seconds + minutes * 60;
+  const newDate = new Date(date);
+  newDate.setSeconds(newDate.getSeconds() + totalSeconds);
+  return newDate;
+};
+
+const firstSessionDate = new Date('03/29/2022 08:50:00 GMT');
+const caseCreationDate = new Date('03/31/2022 16:30:00 GMT');
+const secondSessionDate = new Date('04/01/2022 14:14:00 GMT');
+
+const fakeActions = [
```

Comment:
Use descriptive variable names.
T,N,V ain't.

Reply:
sure, also updated the design doc link thanks!